registry/handlers/app.go

```diff
@@ -396,6 +397,9 @@ func (app *App) dispatcher(dispatch dispatchFunc) http.Handler {
 				serveJSON(w, context.Errors)
 				return
 			}
+		} else {
+			catalog := app.registry.Catalog(context, app.driver)
```

Comment:
There is no expense to construct this on every request. Just set it in the same place as context. We'll optimize this during a refactor.